alchemy/src/os/exec.ts

```diff
@@ -242,3 +266,14 @@ export async function exec(
     });
   });
 }
+
+async function hashInputs(cwd: string, patterns: string[]) {
+  const hash = createHash("sha256");
+  for (const pattern of patterns) {
+    for await (const file of glob(pattern, { cwd })) {
+      const content = await readFile(join(cwd, file), "utf-8");
+      hash.update(content);
+    }
+  }
```

Comment:
This looks sequential. Should we instead use `await Promise.all` to compute the hashes of all files in parallel, then order by path and compute a single hash from the remaining hashes? That would be faster and maybe more deterministic?

Reply:
Good catch! In my experience computing sequentially seems deterministic (obviously raw `Promise.all` isn't, learned that the hard way 😅), but you're right that computing in parallel would be faster. Just pushed an update to do just that.

Reply:
Also side note: the type signature for `glob` ostensibly accepts an array or a string, but Node.js throws if you actually pass it an array (Bun doesn't seem to mind). That's why I processed each glob separately.

Reply:
That's interesting, thanks for being mindful about runtime compatibility!

Reply:
If the globs overlap, could the same file end up being in the files list multiple times? Should we make sure to de-dupe?